serverless/src/api/municipalities/createMunicipalitiesStats/index.js

```diff
@@ -0,0 +1,189 @@
+/**
+ * This endpoint is used to compute stats for municipalities and save them
+ * as a json file on the server.
+ * */
+
+const AWS = require('aws-sdk');
+
+const config = { region: 'eu-central-1' };
+const s3 = new AWS.S3(config);
+const bucket = 'xbge-municipalities-stats';
+const stage = process.env.STAGE;
+
+const { errorResponse } = require('../../../shared/apiResponse');
+const {
+  getAllMunicipalitiesWithUsers,
+  getAllMunicipalities,
+} = require('../../../shared/municipalities');
+const { getMunicipalityGoal } = require('../../../shared/utils');
+
+const timePassed = 6 * 60 * 60 * 1000;
+const amountOfWins = 15;
+const amountOfNewcomers = 15;
+const amountOfChanged = 10;
+const changeThresholds = { absolute: 2000, relative: 30, population: 40000 };
+
+module.exports.handler = async event => {
+  try {
+    // No query param was passed, therefore we get all municipalities for which people
+    // have already signed up and compute the stats
+    const userMuncipality = await getAllMunicipalitiesWithUsers();
+
+    const statsWithAllMunicipalities = await computeStats({
+      userMuncipality,
+      shouldSendAllMunicipalities: true,
+    });
+
+    const statsWithEvents = await computeStats({
+      userMuncipality,
+      shouldSendAllMunicipalities: false,
+    });
+
+    await saveJson(statsWithAllMunicipalities, 'statsWithAll.json');
+    await saveJson(statsWithEvents, 'statsWithEvents.json');
+
+    return event;
+  } catch (error) {
+    console.log('error creating stats for places', error);
+    return errorResponse(500, 'Error while getting stats for places', error);
+  }
+};
+
+const computeStats = async ({
+  userMuncipality,
+  shouldSendAllMunicipalities,
+}) => {
+  const date = new Date();
+  const wins = [];
+  let newcomers = [];
+  let relativeChangers = [];
+  let absoluteChangers = [];
+  const municipalitiesWithUsers = [];
+  console.time('municipalities');
```

Comment:
Timing maybe not necessary in production.

Reply:
Haha, true :D